libcrux-ml-kem/src/utils.rs

```diff
@@ -8,9 +8,16 @@
 #[cfg_attr(hax, hax_lib::requires(
     slice.len() <= LEN
 ))]
+#[cfg_attr(hax, hax_lib::ensures(|result|
+    fstar!("$result == Seq.append $slice (Seq.create (v $LEN - v (${slice.len()})) 0uy)")))]
 pub(crate) fn into_padded_array<const LEN: usize>(slice: &[u8]) -> [u8; LEN] {
     let mut out = [0u8; LEN];
     out[0..slice.len()].copy_from_slice(slice);
+    hax_lib::fstar!("assert (Seq.slice out 0 (Seq.length slice) == slice)");
+    hax_lib::fstar!("assert (Seq.slice out (Seq.length slice) (v v_LEN) == Seq.slice (Seq.create (v v_LEN) 0uy) (Seq.length slice) (v v_LEN))");
+    hax_lib::fstar!("assert (forall i. i < Seq.length slice ==> Seq.index out i == Seq.index slice i)");
+    hax_lib::fstar!("assert (forall i. (i >= Seq.length slice && i < v v_LEN) ==> Seq.index out i == Seq.index (Seq.slice out (Seq.length slice) (v v_LEN)) (i - Seq.length slice))");
+    hax_lib::fstar!("Seq.lemma_eq_intro out (Seq.append slice (Seq.create (v v_LEN - Seq.length slice) 0uy))");
```

Comment:
I would like to switch to a style where we call `fstar_assert` and `fstar_lemma` so that we can get some useful stats out of the proofs. For now, I am using one per line with a view to eventually transforming them.